### PR TITLE
[Refactor] [Diffusion] Refactor custom ops

### PR DESCRIPTION
**Pull request by Makcum888e**
## Motivation

Different hardware need to import different modules for custom op implementation. This PR moves hardware specific implementation to hardware related files. It can decrease needs in `is_npu`, `is_cuda` in custom ops. It also can improve CI: with right filtering changes in hardware specific files will not trigger CI for other hardwares.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/hardware_backend/cuda/layernorm.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        residual_shape = residual.shape
+        residual = residual.view(-1, shape[-1])
+
+    if x.dtype == torch.float:
+        # fp32
+        out = rms_norm_fn(
+            x, self.weight, bias=None, residual=residual, eps=self.variance_epsilon
+        )
+    elif self.variance_size_override is not None:
+        return self.forward_native(x, residual)
```
[the remark this answers is not in the document]
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/hardware_backend/cuda/layernorm.py]: fixed

File: python/sglang/multimodal_gen/runtime/hardware_backend/cuda/layernorm.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    shape = x.shape
+    x = x.view(-1, self.hidden_size)
+    return (
+        self.norm_infer(
+            x.view(-1, self.hidden_size),
+            self.weight,
+            self.bias,
+            eps=self.eps,
+            is_rms_norm=False,
+        )
```
[the remark this answers is not in the document]
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/hardware_backend/cuda/layernorm.py]: fixed

File: python/sglang/multimodal_gen/runtime/backend/native/rotary_embedding.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        if is_neox_style:
+            x1, x2 = torch.chunk(x, 2, dim=-1)
+        else:
+            x1 = x[..., ::2]
+            x2 = x[..., 1::2]
+        o1 = (x1.float() * cos - x2.float() * sin).type_as(x)
+        o2 = (x2.float() * cos + x1.float() * sin).type_as(x)
+        return torch.cat((o1, o2), dim=-1)
+    else:
+        return apply_rotary_embedding(x, cos, sin, interleaved)
```
[the remark this answers is not in the document]
Comment (Makcum888e) [reply python/sglang/multimodal_gen/runtime/backend/native/rotary_embedding.py]: can be refactored in separated PR

**Comment by hnyls2002** [comment]
Hi @Makcum888e, this draft has been inactive for a few months and the diffusion backend layout has since been reorganized (for example multimodal_gen/runtime/backend/cuda/activation.py no longer exists on main), so the refactor would need substantial reworking. Closing as stale; please reopen a rebased version if you'd like to continue.
